adcc/test_AdcMatrix.py

```diff
@@ -214,6 +216,72 @@ def test_matvec_adc2(self):
         diffv = resv.ph - refv.ph
         assert diffv.dot(diffv) < 1e-12
 
+    def test_extra_term(self):
+        ground_state = adcc.LazyMp(cache.refstate["h2o_sto3g"])
+        matrix_adc1 = adcc.AdcMatrix("adc1", ground_state)
+        with pytest.raises(TypeError):
+            matrix_adc1 += 42
+        matrix = adcc.AdcMatrix("adc2", ground_state)
+
+        with pytest.raises(TypeError):
+            adcc.AdcMatrix("adc2", ground_state,
+                           diagonal_precomputed=42)
+        with pytest.raises(ValueError):
+            adcc.AdcMatrix("adc2", ground_state,
+                           diagonal_precomputed=matrix.diagonal() + 42)
+
+        shift = -0.3
+        shifted = AdcMatrixShifted(matrix, shift)
+        # TODO: need to do this to differentiate between
+        # diagonals for ph and pphh
+        # if we just pass numbers, i.e., shift
+        # we get 2*shift on the diagonal :(
```

Comment:
I don't get this comment. Can you explain more?

Reply:
Yeah sure, it's because the diagonal is constructed by
```
other_diagonal = sum(bl.diagonal for bl in other.blocks.values() if bl.diagonal)
```

So, if you want the diagonal contribution from `__shift_ph` and `__shift_pphh`, both have in theory `shift` in the `diagonal` attribute of their `AdcBlock`. But if you don't represent this as an `AmplitudeVector`, the above code will
result in `other_diagonal = shift + shift`, i.e., just adding the two floats.

Reply:
Ah ok now I get it. The issue I was struggeling is the structure of the test. That comment actually refers to something you do much down the line. Could you move the "raises" tests commonly to the beginning or the end to keep the successful test code closer together?

Reply:
Moved them all together except for one, which required `extra` to be present.